tests/defer/tests.py

```diff
@@ -326,3 +327,28 @@ def test_only_select_related_raises_invalid_query(self):
         )
         with self.assertRaisesMessage(FieldError, msg):
             Primary.objects.only("name").select_related("related")[0]
+
+
+class DeferredRelationTests(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.secondary = s = Secondary.objects.create(first="a", second="b")
+        cls.primary = PrimaryOneToOne.objects.create(
+            name="Bella", value="Baxter", related=s
+        )
+
+    def test_defer_does_not_clear_cached_relations(self):
+        obj = Secondary.objects.defer("first").get(id=self.secondary.id)
+        with self.assertNumQueries(1):
+            repr(obj.primary_o2o)
```

Comment:
I believe all the `repr()` calls can be dropped:

```suggestion
            obj.primary_o2o
```